source/articles/2015/save-money-and-be-happier-by-updating-your-gems-every-monday-morning.md

```diff
@@ -0,0 +1,256 @@
+this kind of thing.
+
+[all sorts]: https://appcanary.com/
+[paid services]: https://www.datadoghq.com/lpg/
+
+Great! We're covered, right?
+
+Sorta. This will check the database
+of _reported_ security issues
+which is some subset of released security patches.
+I don't blame gem maintainers for this:
+again, they're working in their spare time
+for the greater good.
+Expecting them to keep up to date
+with more admin activities
+isn't really fair.
+
+So what can we do?
+
+## A Path Forward
+
+I started doing something on projects recently
+which colleagues either don't care about
+or appreciate me doing:
+
+Monday morning,
+I take 15 minutes
+and run `bundle update`.
+
+Then I open a PR with the changes,
+and hopefully deploy within the hour
+once I get code review.
+
+That's it.
+
+The changes are very often quite minor
+and when they're not,
+I look at the gem's release notes
+and figure out what the changes are.
+If it's a big change,
+say new deprecations or
+breaking changes,
+I fix them in about 30 minutes.
+People are pretty good about [SemVer].
+
+[SemVer]: http://semver.org/
+
+> "HAHA, oh great plan but we are _way_ out of date already and can't just jump right up to current."
+
+Of course. So here's what you do:
+Go into the lockfile (`Gemfile.lock`)
+and figure out what the specific versions
+are of things which are really far out of date.
+For example, let's say `huge_bouncer`
+is now at version 2.0.7,
+but we don't know what we are locked at:
+
+```ruby
+# Gemfile.lock
+# ...
+  huge_bouncer (1.2.4)
+    bcrypt
+    email_validator (~> 1.4)
+    rails (>= 3.1)
+```
+
+We can use the [twiddle-waka] operator to say
+"when updating this gem,
+only do so to the next major version,"
+like so:
+
+[twiddle-waka]: https://robots.thoughtbot.com/rubys-pessimistic-operator
+
+```ruby
+# Gemfile
+gem 'huge_bouncer', '~> 1.2.4'
+```
+
+Okay so now we are not moving that much,
+and can do a safe `bundle_update`
+without moving the versions too much.
+In this case, let's say we get up to 1.2.8.
+This is an improvement!
+
+Then, when we next have time,
+we can loosen the restrictions a bit,
+`bundle update` again,
+and see if all is okay.
+
+Remember, we want to make and deploy
+small changes, so don't get all crazy.
+
+## Wait, What About Robots Again?
+
+Shouldn't you build or pay for some service
+which does this for you?
+
+*NO*.
+
+You are including a heck-ton of other code
+and you should be aware of
+how it is changing.
+Maybe not in minute detail,
+but it's a lot of code
+which you are shipping,
+so you should read the label.
+
+
+## The Money Saved
+
+No one will care if you do this
+every Monday morning
+but you are saving your company
+lots of money.
+
+I asked some friends
+to estimate the cost of
+getting totally current with their gems
+at their company or clients
+and here's what they said:
+
+* Upgrading Rails 2 to 3 is about 16+
+developer-weeks, so about a $100k problem
+* Upgrading Rails 3 to 4 is about half of that or less
```

Comment:
I think the ending to this second bullet part is confusing. I think saying about half the time would be sufficient. If you're saying half of 16 weeks, I don't think it can be misinterpreted as "days".